backend/app/core/provider_credentials.py

```
@@ -0,0 +1,165 @@
+    config.pop("credential_configured", None)
+    token_supplied = TOKEN_KEY in config
+    token = config.pop(TOKEN_KEY, None)
+    if token is not None and not isinstance(token, str):
+        raise ValueError("provider token must be a string")
+
+    if not token_supplied and current:
+        _preserve_credential(task_provider, current, config)
+        return config
+
+    normalized_token = token.strip() if isinstance(token, str) else ""
+    if normalized_token and normalized_token != "***":
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "normalized_token"

(S105)